### PR TITLE
feat(dashboard_rbac): roles api for fetching roles and filtering by name

### PR DESCRIPTION
**Pull request by amitmiran137**
### SUMMARY
Api will be used in the UI for fetching available roles for the user to choose

### ADDITIONAL INFORMATION

- [X] Introduces new feature or API
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/roles/api.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    allow_browser_login = True
+    include_route_methods = {
+        RouteMethod.GET,
+        RouteMethod.GET_LIST,
+    }
+
+    show_columns = [
+        "id",
+        "name",
+    ]
```
Comment (ktmud) [inline superset/roles/api.py:45]: I feel we will eventually need a `description` field name, but that is another topic.

**Review by ktmud — APPROVED** [review]
LGTM

**Review by dpgaspar — COMMENTED** [review]
@amitmiran137 

Can you leverage the currently available `.../related` endpoints to fetch the roles to the UI. I think it better for several reasons:
- This is already the pattern to fetch data for UI dropdowns
- A full implementation with `BaseSupersetModelRestApi` will eventually happen, but to implement a full UI for roles and permissions

Does this make sense to you?

**Comment by amitmiran137** [comment]
I wasn't familer with this ability .
So you're saying dashboard Ali should have related config to fetch roles related to a dashboard or just any role ?
Bc in this case we need to be able.to show all roles and not just related ones 

Can you provide an example ?

**Comment by dpgaspar** [comment]
@amitmiran137 

To enable the new field on the `related` endpoint (add roles): https://github.com/apache/superset/blob/master/superset/dashboards/api.py#L200

You can set a default sort also:
https://github.com/apache/superset/blob/master/superset/dashboards/api.py#L192

**Comment by amitmiran137** [comment]
> @amitmiran137 
> 
> To enable the new field on the `related` endpoint (add roles): https://github.com/apache/superset/blob/master/superset/dashboards/api.py#L200
> 
> You can set a default sort also:
> https://github.com/apache/superset/blob/master/superset/dashboards/api.py#L192
> 

In the UI we will need to enable the user to choose roles from the full roles list, unrelated to any dashboard . 
What you're offering won't help

**Comment by dpgaspar** [comment]
@amitmiran137 

The `related` name may be a bit misleading, it will fetch all the roles to populate a dropdown for example

**Comment by ktmud** [comment]
@dpgaspar shouldn't we limit `/related` to real associated entities and run the search filters from a standalone `/users` or `/roles` API?

**Pull request closed without merging**